app/controllers/api/v1/facts_controller.rb

```diff
@@ -32,8 +32,18 @@ def create
 
   # PUT /members/:member_id/facts/:id
   def update
-    @fact.fact_text = params[:fact_text]
-    @fact.likes = params[:likes]
+    prev_fact_text = @fact.fact_text
+    prev_likes = @fact.likes
+    if fact_params[:fact_text]
+      @fact.fact_text = fact_params[:fact_text]
+    else
+      @fact.fact_text = prev_fact_text
+    end
+    if fact_params[:likes]
+      @fact.likes = fact_params[:likes]
+    else
+      @fact.likes = prev_likes
+    end
     if @fact.save
```

Comment:
Your solution for the current schema of the Fact model, so that's good. However, it's not making full use of the "strong parameters" that were taught in [Assignment 7.1](https://github.com/Code-the-Dream-School/R7-forum/blob/main/lessons/Assignment-7_1-Active-Record-Associations.md#step-3-creating-routes-and-the-post-controller). It's not as scalable because it requires `prev` and `if` logic for every attribute. Models with over 10 attributes aren't uncommon in the real world, and that would get quite messy. Here's a solution that I'd recommend:

```suggestion
    if @fact.update(fact_params)
```